### ISSUE TITLE
"Unnecessary arguments" check misinterprets magic constants

### ISSUE THREAD

**Issue by smalyshev** [kalessil/phpinspectionsea#750]
on this code:
```php
function logme($msg, $place = __METHOD__) {
	echo "$place: $msg\n";
}

function logit() {
	logme('test');
	logme('test', __METHOD__);
}
logit();
```

I get warning "argument can be safely dropped" on the second call to logme(). Obviously it is not true - the value of `__METHOD__` is different at the call site and in the default definition. However, the checker thinks the value of `__METHOD__` is the same and can be dropped.

**Comment by smalyshev** [comment]
plugin version:  2.3.15.1

**Comment by rentalhost** [comment]
Make sense, we should consider magic constants here.

Easy way: just ignore magic constants.
Better way: check if magic constant is used on same scope (and have the same value). I guess that this is hard to implement.

**Comment by smalyshev** [comment]
I think probably just ignoring magics would be good enough.

**Comment by kalessil** [comment]
Duplicates #733 (which has been fixed and will be in the next release)
